core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -3741,7 +3741,17 @@ class LogTest {
     assertEquals(new AbortedTransaction(pid, 0), fetchDataInfo.abortedTransactions.get.head)
   }
 
- private def allAbortedTransactions(log: Log) = log.logSegments.flatMap(_.txnIndex.allAbortedTxns)
+  @Test
+  def testLoadPartitionDirWithNoSegmentsShouldNotThrow() {
+    val dirName = Log.logDeleteDirName(new TopicPartition("foo", 3))
+    val logDir = new File(tmpDir, dirName)
+    logDir.mkdirs()
+    val logConfig = LogTest.createLogConfig()
+    // There was a regression in 2.2.1 which threw an NPE
```

Comment:
nit: I don't think we need to mention this in the code. We have the git history and JIRA already.